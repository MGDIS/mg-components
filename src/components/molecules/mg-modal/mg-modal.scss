.mg-modal {
  display: flex;
  justify-content: center;
  align-items: center;
  background-color: hsla(var(--color-dark), 0.3); // TODO: blur
  position: fixed;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  z-index: 1000;

  &__dialog {
    display: flex;
    flex-direction: column;
    row-gap: 3rem;
    padding: 1.5rem 2.5rem;
    border-radius: var(--mg-modal-border-radius);
    background-color: hsla(var(--color-light));
    box-shadow: 0 0.3rem 0.8rem 0 hsla(var(--color-dark), 0.1);
    max-width: 60rem;
  }

  &__header {
    margin-bottom: 0.5rem;
  }

  &__title {
    font-family: var(--font-family-heading);
    font-size: var(--mg-modal-title-font-size);
    font-weight: 600;
    margin: 0;
  }

  &__close-button {
    float: right;
    margin-top: calc((var(--default-size) - (var(--mg-modal-title-font-size) * var(--line-height))) / 2 * -1); //-6px;
    margin-right: calc((var(--default-size) - var(--mg-icon-regular-size)) / 2 * -1); //-10px;
  }

  .mg-modal__content ::slotted(*) {
    font-size: var(--mg-modal-content-font-size);
  }

  &.mg-modal--hide {
    display: none;
  }
}
