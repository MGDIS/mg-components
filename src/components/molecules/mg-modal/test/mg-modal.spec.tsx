import { h } from '@stencil/core';
import { newSpecPage } from '@stencil/core/testing';
import { MgButton } from '../../../atoms/mg-button/mg-button';
import { MgModal } from '../mg-modal';

const getPage = (args, slots?) =>
  newSpecPage({
    components: [MgModal, MgButton],
    template: () => (
      <mg-modal {...args}>
        {slots?.content && (
          <p slot="content">
            <strong>Strong</strong> content!
          </p>
        )}
        {slots?.actions && (
          <div slot="actions" class="mg-group-elements mg-group-elements--align-right">
            <mg-button identifier="identifier">Primary</mg-button>
            <mg-button variant="secondary" identifier="identifier">
              Secondary
            </mg-button>
          </div>
        )}
      </mg-modal>
    ),
  });

describe('mg-modal', () => {
  describe.each([undefined, { content: true }, { actions: true }, { content: true, actions: true }])('Should render a modal', slots => {
    test.each([
      { modalTitle: 'Modal Title', identifier: 'identifier' },
      { modalTitle: 'Modal Title', identifier: 'identifier', closeButton: true },
      { modalTitle: 'Modal Title', identifier: 'identifier', closeButton: true, hide: true },
      { modalTitle: 'Modal Title', identifier: 'identifier', closeButton: true, lang: 'fr' },
      { modalTitle: 'Modal Title', identifier: 'identifier', closeButton: true, lang: 'xx' },
    ])('with args %s', async args => {
      const { root } = await getPage({ ...args }, slots);
      expect(root).toMatchSnapshot();
    });
  });

  test.each(['', ' ', undefined])('Should not render with invalid modalTitle property: %s', async modalTitle => {
    // TODO on 5.0.0 move back to throw new Error test (replace content with commented test)
    // expect.assertions(1);
    // try {
    //   await getPage({ modalTitle });
    // } catch (err) {
    //   expect(err.message).toMatch('<mg-modal> prop "modalTitle" is required.');
    // }
    console.error = jest.fn();
    await getPage({ modalTitle });
    expect(console.error).toHaveBeenCalledWith('<mg-modal> prop "modalTitle" is required.');
  });

  describe('navigation', () => {
    test('should hide panel with button', async () => {
      const page = await getPage({ modalTitle: 'Modal Title', identifier: 'identifier', closeButton: true });
      const closeButton = page.root.shadowRoot.querySelector('#identifier-close-button');

      expect(page.root).toMatchSnapshot();

      const spy = jest.spyOn(page.rootInstance.componentHide, 'emit');

      closeButton.dispatchEvent(new CustomEvent('click', { bubbles: true }));
      await page.waitForChanges();

      expect(page.root).toMatchSnapshot();
      expect(spy).toHaveBeenCalled();
    });
    test.each([true, false])('should hide panel with escape keyboard', async closeButton => {
      const page = await getPage({ modalTitle: 'Modal Title', identifier: 'identifier', closeButton });
      expect(page.root).toMatchSnapshot();

      const spy = jest.spyOn(page.rootInstance.componentHide, 'emit');

      page.win.dispatchEvent(new KeyboardEvent('keydown', { key: 'Escape' }));
      await page.waitForChanges();

      expect(page.root).toMatchSnapshot();
      if (closeButton) {
        expect(spy).toHaveBeenCalled();
      } else {
        expect(spy).not.toHaveBeenCalled();
      }
    });
  });
});
