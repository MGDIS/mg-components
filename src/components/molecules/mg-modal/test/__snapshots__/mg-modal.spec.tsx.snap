// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`mg-modal Should render a modal with args {
  modalTitle: 'Modal Title',
  identifier: 'identifier',
  closeButton: true
} 1`] = `
<mg-modal>
  <mock:shadow-root>
    <section aria-labelledby="identifier-title" aria-modal="true" class="mg-modal" id="identifier" role="dialog" tabindex="-1">
      <article class="mg-modal__dialog">
        <header class="mg-modal__header">
          <span class="mg-modal__close-button">
            <mg-button is-icon="">
              <!---->
              <button aria-disabled="false" aria-label="Close modal" class="mg-button mg-button--flat mg-button--icon" id="identifier-close-button">
                <div class="mg-button__content">
                  <mg-icon icon="cross"></mg-icon>
                </div>
              </button>
            </mg-button>
          </span>
          <h1 class="mg-modal__title" id="identifier-title">
            Modal Title
          </h1>
        </header>
      </article>
    </section>
  </mock:shadow-root>
</mg-modal>
`;

exports[`mg-modal Should render a modal with args {
  modalTitle: 'Modal Title',
  identifier: 'identifier',
  closeButton: true
} 2`] = `
<mg-modal>
  <mock:shadow-root>
    <section aria-labelledby="identifier-title" aria-modal="true" class="mg-modal" id="identifier" role="dialog" tabindex="-1">
      <article class="mg-modal__dialog">
        <header class="mg-modal__header">
          <span class="mg-modal__close-button">
            <mg-button is-icon="">
              <!---->
              <button aria-disabled="false" aria-label="Close modal" class="mg-button mg-button--flat mg-button--icon" id="identifier-close-button">
                <div class="mg-button__content">
                  <mg-icon icon="cross"></mg-icon>
                </div>
              </button>
            </mg-button>
          </span>
          <h1 class="mg-modal__title" id="identifier-title">
            Modal Title
          </h1>
        </header>
        <article class="mg-modal__content">
          <slot name="content"></slot>
        </article>
      </article>
    </section>
  </mock:shadow-root>
  <p slot="content">
    <strong>
      Strong
    </strong>
    content!
  </p>
</mg-modal>
`;

exports[`mg-modal Should render a modal with args {
  modalTitle: 'Modal Title',
  identifier: 'identifier',
  closeButton: true
} 3`] = `
<mg-modal>
  <mock:shadow-root>
    <section aria-labelledby="identifier-title" aria-modal="true" class="mg-modal" id="identifier" role="dialog" tabindex="-1">
      <article class="mg-modal__dialog">
        <header class="mg-modal__header">
          <span class="mg-modal__close-button">
            <mg-button is-icon="">
              <!---->
              <button aria-disabled="false" aria-label="Close modal" class="mg-button mg-button--flat mg-button--icon" id="identifier-close-button">
                <div class="mg-button__content">
                  <mg-icon icon="cross"></mg-icon>
                </div>
              </button>
            </mg-button>
          </span>
          <h1 class="mg-modal__title" id="identifier-title">
            Modal Title
          </h1>
        </header>
        <footer class="mg-modal__footer">
          <slot name="actions"></slot>
        </footer>
      </article>
    </section>
  </mock:shadow-root>
  <div class="mg-group-elements mg-group-elements--align-right" slot="actions">
    <mg-button>
      <!---->
      <button aria-disabled="false" class="mg-button mg-button--primary" id="identifier">
        <div class="mg-button__content">
          Primary
        </div>
      </button>
    </mg-button>
    <mg-button>
      <!---->
      <button aria-disabled="false" class="mg-button mg-button--secondary" id="identifier">
        <div class="mg-button__content">
          Secondary
        </div>
      </button>
    </mg-button>
  </div>
</mg-modal>
`;

exports[`mg-modal Should render a modal with args {
  modalTitle: 'Modal Title',
  identifier: 'identifier',
  closeButton: true
} 4`] = `
<mg-modal>
  <mock:shadow-root>
    <section aria-labelledby="identifier-title" aria-modal="true" class="mg-modal" id="identifier" role="dialog" tabindex="-1">
      <article class="mg-modal__dialog">
        <header class="mg-modal__header">
          <span class="mg-modal__close-button">
            <mg-button is-icon="">
              <!---->
              <button aria-disabled="false" aria-label="Close modal" class="mg-button mg-button--flat mg-button--icon" id="identifier-close-button">
                <div class="mg-button__content">
                  <mg-icon icon="cross"></mg-icon>
                </div>
              </button>
            </mg-button>
          </span>
          <h1 class="mg-modal__title" id="identifier-title">
            Modal Title
          </h1>
        </header>
        <article class="mg-modal__content">
          <slot name="content"></slot>
        </article>
        <footer class="mg-modal__footer">
          <slot name="actions"></slot>
        </footer>
      </article>
    </section>
  </mock:shadow-root>
  <p slot="content">
    <strong>
      Strong
    </strong>
    content!
  </p>
  <div class="mg-group-elements mg-group-elements--align-right" slot="actions">
    <mg-button>
      <!---->
      <button aria-disabled="false" class="mg-button mg-button--primary" id="identifier">
        <div class="mg-button__content">
          Primary
        </div>
      </button>
    </mg-button>
    <mg-button>
      <!---->
      <button aria-disabled="false" class="mg-button mg-button--secondary" id="identifier">
        <div class="mg-button__content">
          Secondary
        </div>
      </button>
    </mg-button>
  </div>
</mg-modal>
`;

exports[`mg-modal Should render a modal with args {
  modalTitle: 'Modal Title',
  identifier: 'identifier',
  closeButton: true,
  hide: true
} 1`] = `
<mg-modal>
  <mock:shadow-root>
    <section aria-hidden="" aria-labelledby="identifier-title" aria-modal="true" class="mg-modal mg-modal--hide" id="identifier" role="dialog" tabindex="-1">
      <article class="mg-modal__dialog">
        <header class="mg-modal__header">
          <span class="mg-modal__close-button">
            <mg-button is-icon="">
              <!---->
              <button aria-disabled="false" aria-label="Close modal" class="mg-button mg-button--flat mg-button--icon" id="identifier-close-button">
                <div class="mg-button__content">
                  <mg-icon icon="cross"></mg-icon>
                </div>
              </button>
            </mg-button>
          </span>
          <h1 class="mg-modal__title" id="identifier-title">
            Modal Title
          </h1>
        </header>
      </article>
    </section>
  </mock:shadow-root>
</mg-modal>
`;

exports[`mg-modal Should render a modal with args {
  modalTitle: 'Modal Title',
  identifier: 'identifier',
  closeButton: true,
  hide: true
} 2`] = `
<mg-modal>
  <mock:shadow-root>
    <section aria-hidden="" aria-labelledby="identifier-title" aria-modal="true" class="mg-modal mg-modal--hide" id="identifier" role="dialog" tabindex="-1">
      <article class="mg-modal__dialog">
        <header class="mg-modal__header">
          <span class="mg-modal__close-button">
            <mg-button is-icon="">
              <!---->
              <button aria-disabled="false" aria-label="Close modal" class="mg-button mg-button--flat mg-button--icon" id="identifier-close-button">
                <div class="mg-button__content">
                  <mg-icon icon="cross"></mg-icon>
                </div>
              </button>
            </mg-button>
          </span>
          <h1 class="mg-modal__title" id="identifier-title">
            Modal Title
          </h1>
        </header>
        <article class="mg-modal__content">
          <slot name="content"></slot>
        </article>
      </article>
    </section>
  </mock:shadow-root>
  <p slot="content">
    <strong>
      Strong
    </strong>
    content!
  </p>
</mg-modal>
`;

exports[`mg-modal Should render a modal with args {
  modalTitle: 'Modal Title',
  identifier: 'identifier',
  closeButton: true,
  hide: true
} 3`] = `
<mg-modal>
  <mock:shadow-root>
    <section aria-hidden="" aria-labelledby="identifier-title" aria-modal="true" class="mg-modal mg-modal--hide" id="identifier" role="dialog" tabindex="-1">
      <article class="mg-modal__dialog">
        <header class="mg-modal__header">
          <span class="mg-modal__close-button">
            <mg-button is-icon="">
              <!---->
              <button aria-disabled="false" aria-label="Close modal" class="mg-button mg-button--flat mg-button--icon" id="identifier-close-button">
                <div class="mg-button__content">
                  <mg-icon icon="cross"></mg-icon>
                </div>
              </button>
            </mg-button>
          </span>
          <h1 class="mg-modal__title" id="identifier-title">
            Modal Title
          </h1>
        </header>
        <footer class="mg-modal__footer">
          <slot name="actions"></slot>
        </footer>
      </article>
    </section>
  </mock:shadow-root>
  <div class="mg-group-elements mg-group-elements--align-right" slot="actions">
    <mg-button>
      <!---->
      <button aria-disabled="false" class="mg-button mg-button--primary" id="identifier">
        <div class="mg-button__content">
          Primary
        </div>
      </button>
    </mg-button>
    <mg-button>
      <!---->
      <button aria-disabled="false" class="mg-button mg-button--secondary" id="identifier">
        <div class="mg-button__content">
          Secondary
        </div>
      </button>
    </mg-button>
  </div>
</mg-modal>
`;

exports[`mg-modal Should render a modal with args {
  modalTitle: 'Modal Title',
  identifier: 'identifier',
  closeButton: true,
  hide: true
} 4`] = `
<mg-modal>
  <mock:shadow-root>
    <section aria-hidden="" aria-labelledby="identifier-title" aria-modal="true" class="mg-modal mg-modal--hide" id="identifier" role="dialog" tabindex="-1">
      <article class="mg-modal__dialog">
        <header class="mg-modal__header">
          <span class="mg-modal__close-button">
            <mg-button is-icon="">
              <!---->
              <button aria-disabled="false" aria-label="Close modal" class="mg-button mg-button--flat mg-button--icon" id="identifier-close-button">
                <div class="mg-button__content">
                  <mg-icon icon="cross"></mg-icon>
                </div>
              </button>
            </mg-button>
          </span>
          <h1 class="mg-modal__title" id="identifier-title">
            Modal Title
          </h1>
        </header>
        <article class="mg-modal__content">
          <slot name="content"></slot>
        </article>
        <footer class="mg-modal__footer">
          <slot name="actions"></slot>
        </footer>
      </article>
    </section>
  </mock:shadow-root>
  <p slot="content">
    <strong>
      Strong
    </strong>
    content!
  </p>
  <div class="mg-group-elements mg-group-elements--align-right" slot="actions">
    <mg-button>
      <!---->
      <button aria-disabled="false" class="mg-button mg-button--primary" id="identifier">
        <div class="mg-button__content">
          Primary
        </div>
      </button>
    </mg-button>
    <mg-button>
      <!---->
      <button aria-disabled="false" class="mg-button mg-button--secondary" id="identifier">
        <div class="mg-button__content">
          Secondary
        </div>
      </button>
    </mg-button>
  </div>
</mg-modal>
`;

exports[`mg-modal Should render a modal with args {
  modalTitle: 'Modal Title',
  identifier: 'identifier',
  closeButton: true,
  lang: 'fr'
} 1`] = `
<mg-modal lang="fr">
  <mock:shadow-root>
    <section aria-labelledby="identifier-title" aria-modal="true" class="mg-modal" id="identifier" role="dialog" tabindex="-1">
      <article class="mg-modal__dialog">
        <header class="mg-modal__header">
          <span class="mg-modal__close-button">
            <mg-button is-icon="">
              <!---->
              <button aria-disabled="false" aria-label="Fermer la modale" class="mg-button mg-button--flat mg-button--icon" id="identifier-close-button">
                <div class="mg-button__content">
                  <mg-icon icon="cross"></mg-icon>
                </div>
              </button>
            </mg-button>
          </span>
          <h1 class="mg-modal__title" id="identifier-title">
            Modal Title
          </h1>
        </header>
      </article>
    </section>
  </mock:shadow-root>
</mg-modal>
`;

exports[`mg-modal Should render a modal with args {
  modalTitle: 'Modal Title',
  identifier: 'identifier',
  closeButton: true,
  lang: 'fr'
} 2`] = `
<mg-modal lang="fr">
  <mock:shadow-root>
    <section aria-labelledby="identifier-title" aria-modal="true" class="mg-modal" id="identifier" role="dialog" tabindex="-1">
      <article class="mg-modal__dialog">
        <header class="mg-modal__header">
          <span class="mg-modal__close-button">
            <mg-button is-icon="">
              <!---->
              <button aria-disabled="false" aria-label="Fermer la modale" class="mg-button mg-button--flat mg-button--icon" id="identifier-close-button">
                <div class="mg-button__content">
                  <mg-icon icon="cross"></mg-icon>
                </div>
              </button>
            </mg-button>
          </span>
          <h1 class="mg-modal__title" id="identifier-title">
            Modal Title
          </h1>
        </header>
        <article class="mg-modal__content">
          <slot name="content"></slot>
        </article>
      </article>
    </section>
  </mock:shadow-root>
  <p slot="content">
    <strong>
      Strong
    </strong>
    content!
  </p>
</mg-modal>
`;

exports[`mg-modal Should render a modal with args {
  modalTitle: 'Modal Title',
  identifier: 'identifier',
  closeButton: true,
  lang: 'fr'
} 3`] = `
<mg-modal lang="fr">
  <mock:shadow-root>
    <section aria-labelledby="identifier-title" aria-modal="true" class="mg-modal" id="identifier" role="dialog" tabindex="-1">
      <article class="mg-modal__dialog">
        <header class="mg-modal__header">
          <span class="mg-modal__close-button">
            <mg-button is-icon="">
              <!---->
              <button aria-disabled="false" aria-label="Fermer la modale" class="mg-button mg-button--flat mg-button--icon" id="identifier-close-button">
                <div class="mg-button__content">
                  <mg-icon icon="cross"></mg-icon>
                </div>
              </button>
            </mg-button>
          </span>
          <h1 class="mg-modal__title" id="identifier-title">
            Modal Title
          </h1>
        </header>
        <footer class="mg-modal__footer">
          <slot name="actions"></slot>
        </footer>
      </article>
    </section>
  </mock:shadow-root>
  <div class="mg-group-elements mg-group-elements--align-right" slot="actions">
    <mg-button>
      <!---->
      <button aria-disabled="false" class="mg-button mg-button--primary" id="identifier">
        <div class="mg-button__content">
          Primary
        </div>
      </button>
    </mg-button>
    <mg-button>
      <!---->
      <button aria-disabled="false" class="mg-button mg-button--secondary" id="identifier">
        <div class="mg-button__content">
          Secondary
        </div>
      </button>
    </mg-button>
  </div>
</mg-modal>
`;

exports[`mg-modal Should render a modal with args {
  modalTitle: 'Modal Title',
  identifier: 'identifier',
  closeButton: true,
  lang: 'fr'
} 4`] = `
<mg-modal lang="fr">
  <mock:shadow-root>
    <section aria-labelledby="identifier-title" aria-modal="true" class="mg-modal" id="identifier" role="dialog" tabindex="-1">
      <article class="mg-modal__dialog">
        <header class="mg-modal__header">
          <span class="mg-modal__close-button">
            <mg-button is-icon="">
              <!---->
              <button aria-disabled="false" aria-label="Fermer la modale" class="mg-button mg-button--flat mg-button--icon" id="identifier-close-button">
                <div class="mg-button__content">
                  <mg-icon icon="cross"></mg-icon>
                </div>
              </button>
            </mg-button>
          </span>
          <h1 class="mg-modal__title" id="identifier-title">
            Modal Title
          </h1>
        </header>
        <article class="mg-modal__content">
          <slot name="content"></slot>
        </article>
        <footer class="mg-modal__footer">
          <slot name="actions"></slot>
        </footer>
      </article>
    </section>
  </mock:shadow-root>
  <p slot="content">
    <strong>
      Strong
    </strong>
    content!
  </p>
  <div class="mg-group-elements mg-group-elements--align-right" slot="actions">
    <mg-button>
      <!---->
      <button aria-disabled="false" class="mg-button mg-button--primary" id="identifier">
        <div class="mg-button__content">
          Primary
        </div>
      </button>
    </mg-button>
    <mg-button>
      <!---->
      <button aria-disabled="false" class="mg-button mg-button--secondary" id="identifier">
        <div class="mg-button__content">
          Secondary
        </div>
      </button>
    </mg-button>
  </div>
</mg-modal>
`;

exports[`mg-modal Should render a modal with args {
  modalTitle: 'Modal Title',
  identifier: 'identifier',
  closeButton: true,
  lang: 'xx'
} 1`] = `
<mg-modal lang="xx">
  <mock:shadow-root>
    <section aria-labelledby="identifier-title" aria-modal="true" class="mg-modal" id="identifier" role="dialog" tabindex="-1">
      <article class="mg-modal__dialog">
        <header class="mg-modal__header">
          <span class="mg-modal__close-button">
            <mg-button is-icon="">
              <!---->
              <button aria-disabled="false" aria-label="Close modal" class="mg-button mg-button--flat mg-button--icon" id="identifier-close-button">
                <div class="mg-button__content">
                  <mg-icon icon="cross"></mg-icon>
                </div>
              </button>
            </mg-button>
          </span>
          <h1 class="mg-modal__title" id="identifier-title">
            Modal Title
          </h1>
        </header>
      </article>
    </section>
  </mock:shadow-root>
</mg-modal>
`;

exports[`mg-modal Should render a modal with args {
  modalTitle: 'Modal Title',
  identifier: 'identifier',
  closeButton: true,
  lang: 'xx'
} 2`] = `
<mg-modal lang="xx">
  <mock:shadow-root>
    <section aria-labelledby="identifier-title" aria-modal="true" class="mg-modal" id="identifier" role="dialog" tabindex="-1">
      <article class="mg-modal__dialog">
        <header class="mg-modal__header">
          <span class="mg-modal__close-button">
            <mg-button is-icon="">
              <!---->
              <button aria-disabled="false" aria-label="Close modal" class="mg-button mg-button--flat mg-button--icon" id="identifier-close-button">
                <div class="mg-button__content">
                  <mg-icon icon="cross"></mg-icon>
                </div>
              </button>
            </mg-button>
          </span>
          <h1 class="mg-modal__title" id="identifier-title">
            Modal Title
          </h1>
        </header>
        <article class="mg-modal__content">
          <slot name="content"></slot>
        </article>
      </article>
    </section>
  </mock:shadow-root>
  <p slot="content">
    <strong>
      Strong
    </strong>
    content!
  </p>
</mg-modal>
`;

exports[`mg-modal Should render a modal with args {
  modalTitle: 'Modal Title',
  identifier: 'identifier',
  closeButton: true,
  lang: 'xx'
} 3`] = `
<mg-modal lang="xx">
  <mock:shadow-root>
    <section aria-labelledby="identifier-title" aria-modal="true" class="mg-modal" id="identifier" role="dialog" tabindex="-1">
      <article class="mg-modal__dialog">
        <header class="mg-modal__header">
          <span class="mg-modal__close-button">
            <mg-button is-icon="">
              <!---->
              <button aria-disabled="false" aria-label="Close modal" class="mg-button mg-button--flat mg-button--icon" id="identifier-close-button">
                <div class="mg-button__content">
                  <mg-icon icon="cross"></mg-icon>
                </div>
              </button>
            </mg-button>
          </span>
          <h1 class="mg-modal__title" id="identifier-title">
            Modal Title
          </h1>
        </header>
        <footer class="mg-modal__footer">
          <slot name="actions"></slot>
        </footer>
      </article>
    </section>
  </mock:shadow-root>
  <div class="mg-group-elements mg-group-elements--align-right" slot="actions">
    <mg-button>
      <!---->
      <button aria-disabled="false" class="mg-button mg-button--primary" id="identifier">
        <div class="mg-button__content">
          Primary
        </div>
      </button>
    </mg-button>
    <mg-button>
      <!---->
      <button aria-disabled="false" class="mg-button mg-button--secondary" id="identifier">
        <div class="mg-button__content">
          Secondary
        </div>
      </button>
    </mg-button>
  </div>
</mg-modal>
`;

exports[`mg-modal Should render a modal with args {
  modalTitle: 'Modal Title',
  identifier: 'identifier',
  closeButton: true,
  lang: 'xx'
} 4`] = `
<mg-modal lang="xx">
  <mock:shadow-root>
    <section aria-labelledby="identifier-title" aria-modal="true" class="mg-modal" id="identifier" role="dialog" tabindex="-1">
      <article class="mg-modal__dialog">
        <header class="mg-modal__header">
          <span class="mg-modal__close-button">
            <mg-button is-icon="">
              <!---->
              <button aria-disabled="false" aria-label="Close modal" class="mg-button mg-button--flat mg-button--icon" id="identifier-close-button">
                <div class="mg-button__content">
                  <mg-icon icon="cross"></mg-icon>
                </div>
              </button>
            </mg-button>
          </span>
          <h1 class="mg-modal__title" id="identifier-title">
            Modal Title
          </h1>
        </header>
        <article class="mg-modal__content">
          <slot name="content"></slot>
        </article>
        <footer class="mg-modal__footer">
          <slot name="actions"></slot>
        </footer>
      </article>
    </section>
  </mock:shadow-root>
  <p slot="content">
    <strong>
      Strong
    </strong>
    content!
  </p>
  <div class="mg-group-elements mg-group-elements--align-right" slot="actions">
    <mg-button>
      <!---->
      <button aria-disabled="false" class="mg-button mg-button--primary" id="identifier">
        <div class="mg-button__content">
          Primary
        </div>
      </button>
    </mg-button>
    <mg-button>
      <!---->
      <button aria-disabled="false" class="mg-button mg-button--secondary" id="identifier">
        <div class="mg-button__content">
          Secondary
        </div>
      </button>
    </mg-button>
  </div>
</mg-modal>
`;

exports[`mg-modal Should render a modal with args { modalTitle: 'Modal Title', identifier: 'identifier' } 1`] = `
<mg-modal>
  <mock:shadow-root>
    <section aria-labelledby="identifier-title" aria-modal="true" class="mg-modal" id="identifier" role="dialog" tabindex="-1">
      <article class="mg-modal__dialog">
        <header class="mg-modal__header">
          <h1 class="mg-modal__title" id="identifier-title">
            Modal Title
          </h1>
        </header>
      </article>
    </section>
  </mock:shadow-root>
</mg-modal>
`;

exports[`mg-modal Should render a modal with args { modalTitle: 'Modal Title', identifier: 'identifier' } 2`] = `
<mg-modal>
  <mock:shadow-root>
    <section aria-labelledby="identifier-title" aria-modal="true" class="mg-modal" id="identifier" role="dialog" tabindex="-1">
      <article class="mg-modal__dialog">
        <header class="mg-modal__header">
          <h1 class="mg-modal__title" id="identifier-title">
            Modal Title
          </h1>
        </header>
        <article class="mg-modal__content">
          <slot name="content"></slot>
        </article>
      </article>
    </section>
  </mock:shadow-root>
  <p slot="content">
    <strong>
      Strong
    </strong>
    content!
  </p>
</mg-modal>
`;

exports[`mg-modal Should render a modal with args { modalTitle: 'Modal Title', identifier: 'identifier' } 3`] = `
<mg-modal>
  <mock:shadow-root>
    <section aria-labelledby="identifier-title" aria-modal="true" class="mg-modal" id="identifier" role="dialog" tabindex="-1">
      <article class="mg-modal__dialog">
        <header class="mg-modal__header">
          <h1 class="mg-modal__title" id="identifier-title">
            Modal Title
          </h1>
        </header>
        <footer class="mg-modal__footer">
          <slot name="actions"></slot>
        </footer>
      </article>
    </section>
  </mock:shadow-root>
  <div class="mg-group-elements mg-group-elements--align-right" slot="actions">
    <mg-button>
      <!---->
      <button aria-disabled="false" class="mg-button mg-button--primary" id="identifier">
        <div class="mg-button__content">
          Primary
        </div>
      </button>
    </mg-button>
    <mg-button>
      <!---->
      <button aria-disabled="false" class="mg-button mg-button--secondary" id="identifier">
        <div class="mg-button__content">
          Secondary
        </div>
      </button>
    </mg-button>
  </div>
</mg-modal>
`;

exports[`mg-modal Should render a modal with args { modalTitle: 'Modal Title', identifier: 'identifier' } 4`] = `
<mg-modal>
  <mock:shadow-root>
    <section aria-labelledby="identifier-title" aria-modal="true" class="mg-modal" id="identifier" role="dialog" tabindex="-1">
      <article class="mg-modal__dialog">
        <header class="mg-modal__header">
          <h1 class="mg-modal__title" id="identifier-title">
            Modal Title
          </h1>
        </header>
        <article class="mg-modal__content">
          <slot name="content"></slot>
        </article>
        <footer class="mg-modal__footer">
          <slot name="actions"></slot>
        </footer>
      </article>
    </section>
  </mock:shadow-root>
  <p slot="content">
    <strong>
      Strong
    </strong>
    content!
  </p>
  <div class="mg-group-elements mg-group-elements--align-right" slot="actions">
    <mg-button>
      <!---->
      <button aria-disabled="false" class="mg-button mg-button--primary" id="identifier">
        <div class="mg-button__content">
          Primary
        </div>
      </button>
    </mg-button>
    <mg-button>
      <!---->
      <button aria-disabled="false" class="mg-button mg-button--secondary" id="identifier">
        <div class="mg-button__content">
          Secondary
        </div>
      </button>
    </mg-button>
  </div>
</mg-modal>
`;

exports[`mg-modal navigation should hide panel with button 1`] = `
<mg-modal>
  <mock:shadow-root>
    <section aria-labelledby="identifier-title" aria-modal="true" class="mg-modal" id="identifier" role="dialog" tabindex="-1">
      <article class="mg-modal__dialog">
        <header class="mg-modal__header">
          <span class="mg-modal__close-button">
            <mg-button is-icon="">
              <!---->
              <button aria-disabled="false" aria-label="Close modal" class="mg-button mg-button--flat mg-button--icon" id="identifier-close-button">
                <div class="mg-button__content">
                  <mg-icon icon="cross"></mg-icon>
                </div>
              </button>
            </mg-button>
          </span>
          <h1 class="mg-modal__title" id="identifier-title">
            Modal Title
          </h1>
        </header>
      </article>
    </section>
  </mock:shadow-root>
</mg-modal>
`;

exports[`mg-modal navigation should hide panel with button 2`] = `
<mg-modal>
  <mock:shadow-root>
    <section aria-hidden="" aria-labelledby="identifier-title" aria-modal="true" class="mg-modal mg-modal--hide" id="identifier" role="dialog" tabindex="-1">
      <article class="mg-modal__dialog">
        <header class="mg-modal__header">
          <span class="mg-modal__close-button">
            <mg-button is-icon="">
              <!---->
              <button aria-disabled="false" aria-label="Close modal" class="mg-button mg-button--flat mg-button--icon" id="identifier-close-button">
                <div class="mg-button__content">
                  <mg-icon icon="cross"></mg-icon>
                </div>
              </button>
            </mg-button>
          </span>
          <h1 class="mg-modal__title" id="identifier-title">
            Modal Title
          </h1>
        </header>
      </article>
    </section>
  </mock:shadow-root>
</mg-modal>
`;

exports[`mg-modal navigation should hide panel with escape keyboard 1`] = `
<mg-modal>
  <mock:shadow-root>
    <section aria-labelledby="identifier-title" aria-modal="true" class="mg-modal" id="identifier" role="dialog" tabindex="-1">
      <article class="mg-modal__dialog">
        <header class="mg-modal__header">
          <span class="mg-modal__close-button">
            <mg-button is-icon="">
              <!---->
              <button aria-disabled="false" aria-label="Close modal" class="mg-button mg-button--flat mg-button--icon" id="identifier-close-button">
                <div class="mg-button__content">
                  <mg-icon icon="cross"></mg-icon>
                </div>
              </button>
            </mg-button>
          </span>
          <h1 class="mg-modal__title" id="identifier-title">
            Modal Title
          </h1>
        </header>
      </article>
    </section>
  </mock:shadow-root>
</mg-modal>
`;

exports[`mg-modal navigation should hide panel with escape keyboard 2`] = `
<mg-modal>
  <mock:shadow-root>
    <section aria-hidden="" aria-labelledby="identifier-title" aria-modal="true" class="mg-modal mg-modal--hide" id="identifier" role="dialog" tabindex="-1">
      <article class="mg-modal__dialog">
        <header class="mg-modal__header">
          <span class="mg-modal__close-button">
            <mg-button is-icon="">
              <!---->
              <button aria-disabled="false" aria-label="Close modal" class="mg-button mg-button--flat mg-button--icon" id="identifier-close-button">
                <div class="mg-button__content">
                  <mg-icon icon="cross"></mg-icon>
                </div>
              </button>
            </mg-button>
          </span>
          <h1 class="mg-modal__title" id="identifier-title">
            Modal Title
          </h1>
        </header>
      </article>
    </section>
  </mock:shadow-root>
</mg-modal>
`;

exports[`mg-modal navigation should hide panel with escape keyboard 3`] = `
<mg-modal>
  <mock:shadow-root>
    <section aria-labelledby="identifier-title" aria-modal="true" class="mg-modal" id="identifier" role="dialog" tabindex="-1">
      <article class="mg-modal__dialog">
        <header class="mg-modal__header">
          <h1 class="mg-modal__title" id="identifier-title">
            Modal Title
          </h1>
        </header>
      </article>
    </section>
  </mock:shadow-root>
</mg-modal>
`;

exports[`mg-modal navigation should hide panel with escape keyboard 4`] = `
<mg-modal>
  <mock:shadow-root>
    <section aria-labelledby="identifier-title" aria-modal="true" class="mg-modal" id="identifier" role="dialog" tabindex="-1">
      <article class="mg-modal__dialog">
        <header class="mg-modal__header">
          <h1 class="mg-modal__title" id="identifier-title">
            Modal Title
          </h1>
        </header>
      </article>
    </section>
  </mock:shadow-root>
</mg-modal>
`;
